shim.go

```diff
@@ -20,59 +20,43 @@ import (
 	"os/exec"
 
 	vc "github.com/containers/virtcontainers"
+	"github.com/golang/glog"
 )
 
-// ShimInfo gathers information needed by the shim.
-type ShimInfo struct {
-	Path  string
-	Token string
-	IP    string
-	Port  string
-}
-
-func startShim(pod *vc.Pod) (int, error) {
-	containers := pod.GetContainers()
-
-	if len(containers) != 1 {
-		return -1, fmt.Errorf("Container list from pod is wrong, expecting only one container")
+func startShim(process *vc.Process, url string) (int, error) {
+	if process.Token == "" {
+		return -1, fmt.Errorf("Token cannot be empty")
 	}
-	container := containers[0]
 
-	shimInfo, err := getShimInfo(container)
-	if err != nil {
-		return -1, err
+	if url == "" {
```

Comment:
This should be an error case. We don't want to support starting a shim without a known URL.